app/forms/classification_form_upload.py

```diff
@@ -0,0 +1,58 @@
+from typing import List
+
+import PIL
+import magic
+
+import starlette.datastructures
+from fastapi import Request, UploadFile
+
+from app.config import Configuration
+
+
+# https://fastapi.tiangolo.com/tutorial/request-forms-and-files/
+class ClassificationFormUpload:
+    _MAGIC_BYTES_TO_READ = 2048  # magic package documentation on GitHub suggests reading at least 2048 bytes.
+
+    def __init__(self, request: Request) -> None:
+        self.request: Request = request
+        self.errors: List = []
+        self.image_file: UploadFile
+        self.image_id: str
+        self.model_id: str
+
+    async def load_data(self):
+        form = await self.request.form()
+        self.image_file = form.get("immagine")
+        self.model_id = form.get("model_id")
+        self.image_id = self.image_file.filename
+
+    async def is_valid(self):
+        # FastAPI bug https://github.com/tiangolo/fastapi/discussions/9705
+        if not self.image_file or not isinstance(self.image_file, starlette.datastructures.UploadFile) \
+                or not self.image_file.filename.endswith(".JPEG"):  # utils.list_images() accepts only '.JPEG' extension
+            self.errors.append("A valid .JPEG image is required (check file extension, it must be uppercase too!)")
```

Comment:
Is this (strong) constraint still required?

Reply:
That, plus the reason mentioned in the comment next to the lines of code.